### PR TITLE
sticky keys refactor

### PR DESCRIPTION
**Pull request by okke-formsma**
This PR contains a number of code readabilty improvements and renames, and does not change any functionality.

### CODE REVIEW

**Comment by okke-formsma** [comment]
@innovaker not a bug, this is just refactors, but depend on #449

**Comment by okke-formsma** [comment]
I rebased the commits so it's only a few simple refactors.

File: app/src/behaviors/behavior_sticky_key.c [hunk 1]
```diff
@@ -21,71 +21,70 @@
...
+active_sticky_key active_sticky_keys[ZMK_BHV_STICKY_KEY_MAX_HELD] = {};
 
-static struct active_sticky_key *store_sticky_key(u32_t position, u32_t param1, u32_t param2,
-                                                  const struct behavior_sticky_key_config *config) {
+static active_sticky_key *store_sticky_key(u32_t position, u32_t param1, u32_t param2,
+                                           const struct behavior_sticky_key_config *config) {
     for (int i = 0; i < ZMK_BHV_STICKY_KEY_MAX_HELD; i++) {
-        if (active_sticky_keys[i].position != ZMK_BHV_STICKY_KEY_POSITION_NOT_USED ||
-            active_sticky_keys[i].timer_is_cancelled) {
+        active_sticky_key *sticky_key = &active_sticky_keys[i];
```
Comment (joelspadin) [inline app/src/behaviors/behavior_sticky_key.c:55]:
```suggestion
        active_sticky_key * const sticky_key = &active_sticky_keys[i];
```

Minor (and if you do accept this, clang-format it first), but putting a `const` after the `*` ensures that `sticky_key` can't be reassigned to point to something else later.

File: app/src/behaviors/behavior_sticky_key.c [hunk 2]
```diff
@@ -21,71 +21,70 @@
...
     bool timer_started;
+    bool timer_cancelled;
     s64_t release_at;
     struct k_delayed_work release_timer;
-    bool timer_is_cancelled;
     // usage page and keycode for the key that is being modified by this sticky key
     u8_t modified_key_usage_page;
     u32_t modified_key_keycode;
-};
+} active_sticky_key;
```
[the remark this answers is not in the document]
Comment (okke-formsma) [reply app/src/behaviors/behavior_sticky_key.c]: I've changed it to simply `sticky_key_t`, as it's the main type that's used in this behavior. I wonder why the main codebase shies away from `typedefs` and writes out `struct this_or_that` everywhere. 

@petejohanson it a deliberate choice to stay away from typedefs and write out `struct` ?

**Comment by okke-formsma** [comment]
Thanks for the review @joelspadin

**Comment by okke-formsma** [comment]
rebased on 2.4.0

File: app/src/behaviors/behavior_sticky_key.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (petejohanson) [reply app/src/behaviors/behavior_sticky_key.c]: @okke-formsma Following Zephyr style. I don't have a strong opinion one way or another, mostly just shooting for consistency.

[the remark this answers is not in the document]
Comment (okke-formsma) [reply app/src/behaviors/behavior_sticky_key.c]: zephyr defers to kernel coding guidelines: https://www.kernel.org/doc/html/v4.10/process/coding-style.html

> In general, a pointer, or a struct that has elements that can reasonably be directly accessed should never be a typedef.

I've removed the typedef.

**Review by petejohanson — APPROVED** [review]

**Pull request merged**